instance.go

```diff
@@ -372,3 +397,59 @@ func instanceFake(args []string, service *update.Service, out *tabwriter.Writer)
 	<-wait
 	return OK
 }
+
+func bench(ops *uint64, id *uint64) {
+	conf := &serverConfig{
+		server: globalFlags.Server,
+	}
+
+	c := &Client{
+		Id:             "filled-out-below",
+		SessionId:      uuid.New(),
+		Version:        instanceFlags.version,
+		AppId:          instanceFlags.appId.String(),
+		Track:          instanceFlags.groupId.String(),
+		config:         conf,
+		errorRate:      instanceFlags.errorRate,
+		pingsRemaining: instanceFlags.pingOnly,
+		forceUpdate:    instanceFlags.forceUpdate,
+	}
+
+	for {
+		i := atomic.AddUint64(id, 1) // Generate unique new ID
+		c.Id = "deadbeefdeadbeef" + fmt.Sprintf("%016x", i)
+		resp, err := c.MakeRequest("3", "1", true, true)
+		// Log errors because they are not expected
+		if err != nil {
+			c.Log("err: %v\n", err)
+		} else if resp.Apps[0].UpdateCheck.Status != "ok" {
+			c.Log("status: %s\n", resp.Apps[0].UpdateCheck.Status)
+		} else if instanceFlags.verbose {
+			c.Log("update to %v\n", resp.Apps[0].UpdateCheck.Manifest.Version)
+		}
+		// No full update circle, just one request
+		atomic.AddUint64(ops, 1) // Increase global response count
+	}
+}
+
+func instanceBench(args []string, service *update.Service, out *tabwriter.Writer) int {
+	if instanceFlags.appId.Get() == nil || instanceFlags.groupId.Get() == nil || instanceFlags.benchSeconds == 0 {
+		return ERROR_USAGE
+	}
+
+	var ops uint64 // Global response counter
+	var id uint64 // Global ID counter for the second half of the instance ID
+	rand.Seed(time.Now().UnixNano())
+	id = rand.Uint64() // Random start point for uniqueness
+	for i := 0; i < instanceFlags.clientsPerApp; i++ {
+		go bench(&ops, &id)
+	}
+
+	time.Sleep(time.Duration(instanceFlags.benchSeconds) * time.Second)
```

Comment:
I have previously worked on a similar benchmark in another project and instead of sleeping, it was nicer to print the ongoing results so far. i.e. total responses so far, etc.

Reply:
Good idea, updated it